src/v/model/transform.cc

```diff
@@ -307,4 +307,10 @@ transformed_data transformed_data::copy() const {
     return transformed_data(_data.copy());
 }
 
+wasm_binary_iobuf share_wasm_binary(const wasm_binary_iobuf& b) {
+    const auto& underlying = b();
+    return wasm_binary_iobuf(
+      std::make_unique<iobuf>(underlying->share(0, underlying->size_bytes())));
```

Comment:
i'm not fully understanding this. if you are moving the iobuf from one core to another, then the foreign pointer makes sense--the destination core will arrange for the iobuf to be destroyed on the originating core, and don't need to share() anything.

if you sending a zero-copy shared iobuf to the other core, then I'm not entirely sure that is even safe because because the underlying temporary buffers in the iobuf are using reference counting.

Reply:
> I mean it's safe if you don't touch anything or share anything, right?

Yes I think that is true, and in all the cases here that is true.

> Can you move ownership of the iobuf?

I don't think so. During retries leadership could change and the core on which we're executing on.